sql/src/test/java/io/crate/operation/scalar/arithmetic/TrigonometricFunctionTest.java

```diff
@@ -0,0 +1,278 @@
+
+        // TanFunction
+        assertThat((Double) evaluateTan(1.0, DataTypes.DOUBLE), is(1.5574077246549023));
+        assertThat((Double) evaluateTan(2.0F, DataTypes.FLOAT), is(-2.185039863261519));
+        assertThat((Double) evaluateTan(3L, DataTypes.LONG), is(-0.1425465430742778));
+        assertThat((Double) evaluateTan(4, DataTypes.INTEGER), is(1.1578212823495777));
+        assertThat((Double) evaluateTan((short) 5, DataTypes.SHORT), is(-3.380515006246586));
+
+        // AtanFunction
+        assertThat((Double) evaluateAtan(0.1234, DataTypes.DOUBLE), is(0.12277930094473836));
+        assertThat((Double) evaluateAtan(0.4321F, DataTypes.FLOAT), is(0.4078690066146179));
+        assertThat((Double) evaluateAtan(0L, DataTypes.LONG), is(0.0));
+        assertThat((Double) evaluateAtan(1, DataTypes.INTEGER), is(0.7853981633974483));
+        assertThat((Double) evaluateAtan((short) -1, DataTypes.SHORT), is(-0.7853981633974483));
+    }
+
+    @Test
+    public void testEvaluateAsinOnIllegalArgument() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("asin(x): is defined only for values in the range of [-1.0, 1.0]");
+        evaluateAsin(2.0, DataTypes.DOUBLE);
+    }
+
+    @Test
+    public void testEvaluateAcosOnIllegalArgument() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("acos(x): is defined only for values in the range of [-1.0, 1.0]");
+        evaluateAcos(2.0, DataTypes.DOUBLE);
+    }
+
+    @Test
+    public void testEvaluateAtanOnIllegalArgument() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("atan(x): is defined only for values in the range of [-1.0, 1.0]");
+        evaluateAtan(2.0, DataTypes.DOUBLE);
+    }
+
+    @Test
+    public void testEvaluateOnNull() throws Exception {
+        for (DataType type : DataTypes.NUMERIC_PRIMITIVE_TYPES) {
+            // SinFunction
+            assertThat(evaluateSin(null, type), nullValue());
+
+            // AsinFunction
+            assertThat(evaluateAsin(null, type), nullValue());
+
+            // CosFunction
+            assertThat(evaluateCos(null, type), nullValue());
+
+            // AcosFunction
+            assertThat(evaluateAcos(null, type), nullValue());
+
+            // TanFunction
+            assertThat(evaluateTan(null, type), nullValue());
+
+            // AtanFunction
+            assertThat(evaluateAtan(null, type), nullValue());
+        }
+    }
+
+    @Test
+    public void testNormalize() throws Exception {
+        // SinFunction
+        assertNormalize("sin(1.0)", isLiteral(0.8414709848078965, DataTypes.DOUBLE));
+        assertNormalize("sin(cast (2.0 as float))", isLiteral(0.9092974268256817, DataTypes.DOUBLE));
+        assertNormalize("sin(cast (3 as long))", isLiteral(0.1411200080598672, DataTypes.DOUBLE));
```

Comment:
no need to cast as long, number are always return from the sql parser as long or double (pls change at below tests too)